cli/smartcontract/contract_test.go

```diff
@@ -213,6 +203,87 @@ func ToMap(a []testcontract.MyPair) map[int]string {
 `, string(bs))
 }
 
+// updateGoMod updates the go.mod file located in the specified directory.
+// It sets the module name and replaces the neo-go interop package path.
+func updateGoMod(dir, moduleName, neoGoPath string) error {
+	goModPath := filepath.Join(dir, "go.mod")
+	data, err := os.ReadFile(goModPath)
+	if err != nil {
+		return fmt.Errorf("failed to read go.mod: %w", err)
+	}
+
+	i := bytes.IndexByte(data, '\n')
+	if i == -1 {
+		return fmt.Errorf("unexpected go.mod format")
+	}
+
+	updatedData := append([]byte("module "+moduleName), data[i:]...)
+	wd, err := os.Getwd()
+	if err != nil {
+		return fmt.Errorf("failed to get working directory: %w", err)
+	}
+
+	replacementPath := filepath.Join(wd, neoGoPath)
+	updatedData = append(updatedData, "\nreplace github.com/nspcc-dev/neo-go/pkg/interop => "+replacementPath+" \n"...)
+
+	if err := os.WriteFile(goModPath, updatedData, os.ModePerm); err != nil {
+		return fmt.Errorf("failed to write updated go.mod: %w", err)
+	}
+
+	return nil
+}
+
+func TestDynamicWrapper(t *testing.T) {
+	smartcontract.ModVersion = "v0.0.0"
+
+	tmpDir := t.TempDir()
+	e := testcli.NewExecutor(t, true)
+
+	ctrPath := "../smartcontract/testdata"
+
+	verifyHash := testcli.DeployContract(t, e, filepath.Join(ctrPath, "verify.go"), filepath.Join(ctrPath, "verify.yml"), testcli.ValidatorWallet, testcli.ValidatorAddr, testcli.ValidatorPass)
+
+	helperContract := `package testcontract
+
+import (
+	"github.com/nspcc-dev/neo-go/pkg/interop"
+	verify "myimport.com/testcontract/bindings"
+)
+
+func CallVerifyContract(h interop.Hash160) bool{
+	contractInstance := verify.NewContract(h)
+	return contractInstance.Verify()
+}`
+
+	helperDir := filepath.Join(tmpDir, "helper")
```

Comment:
Do you reallyy need this extra `helperDir`? Why can't you use `tmpDi` directly?

Reply:
if I try to init contract in an existing temp directory I get the `A file with given smart-contract name already exists` error

Reply:
Something from the previous run? I won't expect that, but OK.